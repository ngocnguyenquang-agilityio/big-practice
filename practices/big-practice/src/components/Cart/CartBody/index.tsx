// Stores
import { useCart } from '@stores/cart';

// Components
import ProductCartItem from '../ProductCartItem';
import CartEmpty from '../CartEmpty';

const CartBody = () => {
  const { cartItems } = useCart();

  const totalPrice = cartItems.reduce((total, item) => total + item.product.price * item.quantity, 0);

  return cartItems.length > 0 ? (
    <div className='flex h-full flex-col justify-between overflow-hidden p-1'>
      <ul className='flex-grow overflow-auto py-4'>
        {cartItems.map((item) => (
          <li
            key={item.id}
            className='flex w-full flex-col border-b border-neutral-700'
          >
            <ProductCartItem
              id={item.id}
              product={item.product}
              quantity={item.quantity}
            />
          </li>
        ))}
      </ul>
      <div className='py-4 text-sm text-neutral-400'>
        <div className='mb-3 flex items-center justify-between border-b pb-1 border-neutral-700'>
          <p>Taxes</p>
          <p className='text-right text-base text-white'>
            $0.00
            <span className='ml-1 inline'>USD</span>
          </p>
        </div>
        <div className='mb-3 flex items-center justify-between border-b pb-1 pt-1 border-neutral-700'>
          <p>Shipping</p>
          <p className='text-right'>Calculated at checkout</p>
        </div>
        <div className='mb-3 flex items-center justify-between border-b pb-1 pt-1 border-neutral-700'>
          <p>Total</p>
          <p className='text-right text-base text-white'>
            ${totalPrice}
            <span className='ml-1 inline'>USD</span>
          </p>
        </div>
      </div>
    </div>
  ) : (
    <CartEmpty />
  );
};

export default CartBody;
