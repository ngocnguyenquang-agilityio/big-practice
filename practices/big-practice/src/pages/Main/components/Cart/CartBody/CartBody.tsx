// Stores
import { useCartStore } from '@stores/cartStore';

// Libs
import { Link } from 'react-router-dom';

// Components
import CartEmpty from '../CartEmpty/CartEmpty';
import { Button } from '@components/Button/Button';
import ProductCartItemContainer from '../ProductCartItem/ProductCartItem';

// Constants
import { APP_ROUTERS } from '@constants';

const CartBody = () => {
  const { toggleCart, cartItems } = useCartStore();

  const totalPrice = cartItems.reduce((total, item) => total + item.price * item.quantity, 0);

  return cartItems.length > 0 ? (
    <div className='flex h-full flex-col justify-between overflow-hidden p-1'>
      <ul className='flex-grow overflow-auto py-4'>
        {cartItems.map((item) => (
          <li
            key={item.id}
            className='flex w-full flex-col border-b border-neutral-700'
          >
            <ProductCartItemContainer {...item} />
          </li>
        ))}
      </ul>
      <div className='py-4 text-sm text-neutral-400'>
        <div className='mb-3 flex items-center justify-between border-b pb-1 border-neutral-700'>
          <p>Taxes</p>
          <p className='text-right text-base text-white'>
            $0.00
            <span className='ml-1 inline'>USD</span>
          </p>
        </div>
        <div className='mb-3 flex items-center justify-between border-b pb-1 pt-1 border-neutral-700'>
          <p>Shipping</p>
          <p className='text-right'>Calculated at checkout</p>
        </div>
        <div className='mb-3 flex items-center justify-between border-b pb-1 pt-1 border-neutral-700'>
          <p>Total</p>
          <p className='text-right text-base text-white'>
            ${totalPrice}
            <span className='ml-1 inline'>USD</span>
          </p>
        </div>
        <Link to={APP_ROUTERS.CHECKOUT}>
          <Button
            size='xl'
            onClick={toggleCart}
          >
            Proceed to Checkout
          </Button>
        </Link>
      </div>
    </div>
  ) : (
    <CartEmpty />
  );
};

export default CartBody;
