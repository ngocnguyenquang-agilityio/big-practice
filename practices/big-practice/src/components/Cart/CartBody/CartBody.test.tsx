import { render } from '@testing-library/react';

// Components
import CartBody from '.';

describe('CartBody Component', () => {
  test('Should render correctly', () => {
    const comp = render(<CartBody />);
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    const comp = render(<CartBody />);
    expect(comp).toMatchSnapshot();
  });
});
