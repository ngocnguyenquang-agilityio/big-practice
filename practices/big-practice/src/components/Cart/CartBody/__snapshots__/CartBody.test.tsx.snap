// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartBody Component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex h-full flex-col justify-between overflow-hidden p-1"
      >
        <ul
          class="flex-grow overflow-auto py-4"
        >
          <li
            class="flex w-full flex-col border-b border-neutral-700"
          >
            <div
              class="relative flex w-full flex-row justify-between px-1 py-4"
            >
              <div
                class="absolute z-40 -mt-2 ml-[55px]"
              >
                <button
                  class="ease flex h-[17px] w-[17px] items-center justify-center rounded-full bg-neutral-500 transition-all duration-200"
                >
                  <svg
                    aria-hidden="true"
                    class="hover:text-accent-3 mx-[1px] h-4 w-4 text-white dark:text-black"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6 18L18 6M6 6l12 12"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </button>
              </div>
              <a
                class="z-30 flex flex-row space-x-4"
                href=""
              >
                <div
                  class="relative h-16 w-16 cursor-pointer overflow-hidden rounded-md border border-neutral-700 bg-neutral-900 hover:bg-neutral-800"
                >
                  <img
                    class="h-full w-full object-cover"
                    loading="lazy"
                    src="https://demo.vercel.store/_next/image?url=https%3A%2F%2Fcdn.shopify.com%2Fs%2Ffiles%2F1%2F0754%2F3727%2F7491%2Ffiles%2Fkeyboard.png%3Fv%3D1690003507&w=3840&q=75"
                  />
                </div>
                <div
                  class="flex flex-1 flex-col text-base"
                >
                  <span
                    class="leading-tight"
                  >
                    Test Product
                  </span>
                  <p
                    class="text-sm text-neutral-400"
                  >
                    Black / 7 x 9 inch
                  </p>
                </div>
              </a>
              <div
                class="flex h-16 flex-col justify-between"
              >
                <p
                  class="flex justify-end space-y-2 text-right text-sm"
                >
                  20
                  <span
                    class="ml-1 inline"
                  >
                    USD
                  </span>
                </p>
                <div
                  class="ml-auto flex h-9 flex-row items-center rounded-full border border-neutral-700"
                >
                  <button
                    class="text-white text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 py-2 ease flex h-full min-w-[36px] max-w-[36px] flex-none items-center justify-center rounded-full px-2 transition-all duration-200 hover:border-neutral-800 hover:opacity-80 ml-auto"
                  >
                    <svg
                      aria-hidden="true"
                      class="h-4 w-4 dark:text-neutral-500"
                      fill="none"
                      stroke="currentColor"
                      stroke-width="1.5"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M19.5 12h-15"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                    </svg>
                  </button>
                  <p
                    class="w-6 text-center"
                  >
                    <span
                      class="w-full text-sm text-white"
                    >
                      1
                    </span>
                  </p>
                  <button
                    class="text-white text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 py-2 ease flex h-full min-w-[36px] max-w-[36px] flex-none items-center justify-center rounded-full px-2 transition-all duration-200 hover:border-neutral-800 hover:opacity-80"
                  >
                    <svg
                      aria-hidden="true"
                      class="h-4 w-4 dark:text-neutral-500"
                      fill="none"
                      stroke="currentColor"
                      stroke-width="1.5"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 4.5v15m7.5-7.5h-15"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </li>
        </ul>
        <div
          class="py-4 text-sm text-neutral-400"
        >
          <div
            class="mb-3 flex items-center justify-between border-b pb-1 border-neutral-700"
          >
            <p>
              Taxes
            </p>
            <p
              class="text-right text-base text-white"
            >
              $7.00
              <span
                class="ml-1 inline"
              >
                USD
              </span>
            </p>
          </div>
          <div
            class="mb-3 flex items-center justify-between border-b pb-1 pt-1 border-neutral-700"
          >
            <p>
              Shipping
            </p>
            <p
              class="text-right"
            >
              Calculated at checkout
            </p>
          </div>
          <div
            class="mb-3 flex items-center justify-between border-b pb-1 pt-1 border-neutral-700"
          >
            <p>
              Total
            </p>
            <p
              class="text-right text-base text-white"
            >
              $7.00
              <span
                class="ml-1 inline"
              >
                USD
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex h-full flex-col justify-between overflow-hidden p-1"
    >
      <ul
        class="flex-grow overflow-auto py-4"
      >
        <li
          class="flex w-full flex-col border-b border-neutral-700"
        >
          <div
            class="relative flex w-full flex-row justify-between px-1 py-4"
          >
            <div
              class="absolute z-40 -mt-2 ml-[55px]"
            >
              <button
                class="ease flex h-[17px] w-[17px] items-center justify-center rounded-full bg-neutral-500 transition-all duration-200"
              >
                <svg
                  aria-hidden="true"
                  class="hover:text-accent-3 mx-[1px] h-4 w-4 text-white dark:text-black"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6 18L18 6M6 6l12 12"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
            <a
              class="z-30 flex flex-row space-x-4"
              href=""
            >
              <div
                class="relative h-16 w-16 cursor-pointer overflow-hidden rounded-md border border-neutral-700 bg-neutral-900 hover:bg-neutral-800"
              >
                <img
                  class="h-full w-full object-cover"
                  loading="lazy"
                  src="https://demo.vercel.store/_next/image?url=https%3A%2F%2Fcdn.shopify.com%2Fs%2Ffiles%2F1%2F0754%2F3727%2F7491%2Ffiles%2Fkeyboard.png%3Fv%3D1690003507&w=3840&q=75"
                />
              </div>
              <div
                class="flex flex-1 flex-col text-base"
              >
                <span
                  class="leading-tight"
                >
                  Test Product
                </span>
                <p
                  class="text-sm text-neutral-400"
                >
                  Black / 7 x 9 inch
                </p>
              </div>
            </a>
            <div
              class="flex h-16 flex-col justify-between"
            >
              <p
                class="flex justify-end space-y-2 text-right text-sm"
              >
                20
                <span
                  class="ml-1 inline"
                >
                  USD
                </span>
              </p>
              <div
                class="ml-auto flex h-9 flex-row items-center rounded-full border border-neutral-700"
              >
                <button
                  class="text-white text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 py-2 ease flex h-full min-w-[36px] max-w-[36px] flex-none items-center justify-center rounded-full px-2 transition-all duration-200 hover:border-neutral-800 hover:opacity-80 ml-auto"
                >
                  <svg
                    aria-hidden="true"
                    class="h-4 w-4 dark:text-neutral-500"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M19.5 12h-15"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </button>
                <p
                  class="w-6 text-center"
                >
                  <span
                    class="w-full text-sm text-white"
                  >
                    1
                  </span>
                </p>
                <button
                  class="text-white text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 py-2 ease flex h-full min-w-[36px] max-w-[36px] flex-none items-center justify-center rounded-full px-2 transition-all duration-200 hover:border-neutral-800 hover:opacity-80"
                >
                  <svg
                    aria-hidden="true"
                    class="h-4 w-4 dark:text-neutral-500"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 4.5v15m7.5-7.5h-15"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </li>
      </ul>
      <div
        class="py-4 text-sm text-neutral-400"
      >
        <div
          class="mb-3 flex items-center justify-between border-b pb-1 border-neutral-700"
        >
          <p>
            Taxes
          </p>
          <p
            class="text-right text-base text-white"
          >
            $7.00
            <span
              class="ml-1 inline"
            >
              USD
            </span>
          </p>
        </div>
        <div
          class="mb-3 flex items-center justify-between border-b pb-1 pt-1 border-neutral-700"
        >
          <p>
            Shipping
          </p>
          <p
            class="text-right"
          >
            Calculated at checkout
          </p>
        </div>
        <div
          class="mb-3 flex items-center justify-between border-b pb-1 pt-1 border-neutral-700"
        >
          <p>
            Total
          </p>
          <p
            class="text-right text-base text-white"
          >
            $7.00
            <span
              class="ml-1 inline"
            >
              USD
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
